framework/src/actions/SetupMeshAction.C

```diff
@@ -152,6 +153,34 @@ SetupMeshAction::setupMesh(MooseMesh * mesh)
 void
 SetupMeshAction::act()
 {
+  // switch non-file meshes to be a file-mesh if using a pre-split mesh configuration.
+  if (_app.parameters().get<bool>("use_split"))
+  {
+    auto split_file = _app.parameters().get<std::string>("split_file");
+    if (_type != "FileMesh")
+    {
+      if (split_file == "")
+        mooseError("cannot use split mesh with non-file mesh without specifying --split-file on "
+                   "command line");
+
+      _type = "FileMesh";
+      auto new_pars = validParams<FileMesh>();
+      new_pars.set<MeshFileName>("file") = split_file;
+      new_pars.set<MooseApp *>("_moose_app") = _moose_object_pars.get<MooseApp *>("_moose_app");
+      new_pars.set<MooseEnum>("parallel_type") = _moose_object_pars.get<MooseEnum>("parallel_type");
+      new_pars.set<MooseEnum>("parallel_type") = "distributed";
```

Comment:
I don't think this is correct, two sets in a row for the same parameter?

Reply:
Oops